ciao-scheduler/scheduler.go

```diff
@@ -471,104 +471,33 @@ func (sched *ssntpSchedulerServer) StatusNotify(uuid string, status ssntp.Status
 
 type workResources struct {
 	instanceUUID string
-	memReqMB     int
 	diskReqMB    int
-	networkNode  bool
-	physNets     []string
+	requirements payloads.WorkloadRequirements
 }
 
 func (sched *ssntpSchedulerServer) getWorkloadResources(work *payloads.Start) (workload workResources, err error) {
-	// loop the array to find resources
-	for idx := range work.Start.RequestedResources {
-		reqType := work.Start.RequestedResources[idx].Type
-		reqValue := work.Start.RequestedResources[idx].Value
-		reqString := work.Start.RequestedResources[idx].ValueString
-
-		// memory:
-		if reqType == payloads.MemMB {
-			workload.memReqMB = reqValue
-		}
-
-		// network node
-		if reqType == payloads.NetworkNode {
-			wantsNetworkNode := reqValue
-
-			// validate input: requested resource values are always integers
-			if wantsNetworkNode != 0 && wantsNetworkNode != 1 {
-				return workload, fmt.Errorf("invalid start payload resource demand: network_node (%d) is not 0 or 1", wantsNetworkNode)
-			}
-
-			// convert to more natural bool for local struct
-			if wantsNetworkNode == 1 {
-				workload.networkNode = true
-			} else { //wantsNetworkNode == 0
-				workload.networkNode = false
-			}
-
-		}
-
-		// network node physical networks
-		if workload.networkNode {
-			if reqType == payloads.PhysicalNetwork {
-				workload.physNets = append(workload.physNets, reqString)
-			}
-		}
-
-		// etc...
-	}
-
 	// volumes
 	for _, volume := range work.Start.Storage {
 		if volume.Local {
 			workload.diskReqMB += volume.Size * 1024
 		}
 	}
 
-	// validate the found resources
-	if workload.memReqMB <= 0 {
-		return workload, fmt.Errorf("invalid start payload resource demand: mem_mb (%d) <= 0, must be > 0", workload.memReqMB)
-	}
-	if workload.diskReqMB < 0 {
-		return workload, fmt.Errorf("invalid start payload local disk demand: disk MB (%d) < 0, must be >= 0", workload.diskReqMB)
-	}
+	workload.requirements = work.Start.Requirements
 
 	// note the uuid
 	workload.instanceUUID = work.Start.InstanceUUID
 
 	return workload, nil
 }
 
-func networkDemandsSatisfied(node *nodeStat, workload *workResources) bool {
-	if !node.isNetNode {
-		return true
-	}
-
-	var matchedNetworksCount int
-	var requestedNetworksCount int
-
-	for _, requestedNetwork := range workload.physNets {
```

Comment:
Was this stuff simply not used?

Reply:
With the exception of the !node.isNetNode() which is now done in the caller, yep this code was not used.